lmdeploy/pytorch/engine/engine.py

```diff
@@ -371,19 +387,22 @@ def __init__(self,
         misc_config = _build_misc_config(engine_config)
 
         # build model agent
-        self.executor = build_executor(model_path,
-                                       cache_config=cache_config,
-                                       backend_config=backend_config,
-                                       dist_config=dist_config,
-                                       misc_config=misc_config,
-                                       adapters=adapters,
-                                       device_type=engine_config.device_type,
-                                       distributed_executor_backend=engine_config.distributed_executor_backend,
-                                       dtype=engine_config.dtype)
+        self.executor = build_executor(
+            model_path,
+            cache_config=cache_config,
+            backend_config=backend_config,
+            dist_config=dist_config,
+            misc_config=misc_config,
+            adapters=adapters,
+            device_type=engine_config.device_type,
+            distributed_executor_backend=engine_config.distributed_executor_backend,
+            dtype=engine_config.dtype,
+            speculative_config=speculative_config,
```

Comment:
Speculative_config is define outside pytorch engine. It is not a good design to use it in any module besides Engine.